dynamometer-workload/src/test/java/com/linkedin/dynamometer/workloadgenerator/TestWorkloadGenerator.java

```diff
@@ -63,15 +61,40 @@ public void testAuditWorkloadDirectParser() throws Exception {
         TestWorkloadGenerator.class.getClassLoader().getResource("audit_trace_direct").toString();
     conf.set(AuditReplayMapper.INPUT_PATH_KEY, workloadInputPath);
     conf.setLong(AuditLogDirectParser.AUDIT_START_TIMESTAMP_KEY, 60*1000);
-    testAuditWorkload();
+    testAuditWorkloadWithReducer(null);
   }
 
   @Test
   public void testAuditWorkloadHiveParser() throws Exception {
-    String workloadInputPath = TestWorkloadGenerator.class.getClassLoader().getResource("audit_trace_hive").toString();
+    String workloadInputPath =
+        TestWorkloadGenerator.class.getClassLoader().getResource("audit_trace_hive").toString();
+    conf.set(AuditReplayMapper.INPUT_PATH_KEY, workloadInputPath);
+    conf.setClass(AuditReplayMapper.COMMAND_PARSER_KEY, AuditLogHiveTableParser.class, AuditCommandParser.class);
+    testAuditWorkloadWithReducer(null);
+  }
+
+  @Test
+  public void testAuditWorkloadDirectParserWithOutput() throws Exception {
+    String workloadInputPath =
+        TestWorkloadGenerator.class.getClassLoader().getResource("audit_trace_direct").toString();
+    String auditOutputPath = "/reducer_output/trace_output_direct";
+    conf.set(AuditReplayMapper.INPUT_PATH_KEY, workloadInputPath);
+    conf.set(AuditReplayReducer.OUTPUT_PATH_KEY, auditOutputPath);
+    conf.setLong(AuditLogDirectParser.AUDIT_START_TIMESTAMP_KEY, 60*1000);
+    testAuditWorkloadWithReducer(AuditReplayReducer.class);
+    assertTrue(dfs.exists(new Path(auditOutputPath)));
```

Comment:
The tests look great, good use of DRY. But let's not just assert the file exists, let's examine its contents as well.